File: app/src/main/java/net/xcreen/restsms/server/ServerLogging.java
package net.xcreen.restsms.server;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.util.Log;

import java.io.BufferedWriter;
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;

public class ServerLogging {

    private String filepath = null;
    private Context context;

    public ServerLogging(String filepath, Context context){
        this.filepath = filepath + File.separator + "logs";
        this.context = context;
    }

    /**
     * Write Log-Entry to File
     * @param type - String Log-Type (error,warning,info)
     * @param message - String Log-Message
     */
    public void log(String type, String message){
        //Stop processing, when logging is disabled
        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
        if(sharedPref.getBoolean("disable_logging", false)){
            return;
        }
        //Create and get Log-File
        String path = setupLogging();

        //Create Entry
        DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault());
        Date date = new Date();
        String entry = dateFormat.format(date) + " - Type: " + type + " - Message: " + message;

        //Write Log-Output
        switch (type){
            case "error":
                Log.e("Server-Logging", entry);
                break;
            case "warning":
                Log.w("Server-Logging", entry);
                break;
            case "debug":
                Log.d("Server-Logging", entry);
                break;
            case "info":
                Log.i("Server-Logging", entry);
                break;
            default:
                Log.i("Server-Logging", entry);
                break;
        }

        //Write to Log-File
        if(path != null) {
            try (FileWriter fw = new FileWriter(path, true);
                 BufferedWriter bw = new BufferedWriter(fw);
                 PrintWriter out = new PrintWriter(bw)) {
                out.println(entry);
            } catch (IOException ex) {
                ex.printStackTrace();
            }
        }
        else{
            Log.i("Server-Logging", "Failed to write to log!");
        }
    }

    /**
     * Get File for Logging
     * @return String - Log-File-Path
     */
    private String setupLogging(){
        if(filepath != null){
            //Create Filename
            DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd", Locale.getDefault());
            Date date = new Date();
            String filename = dateFormat.format(date) + "-server.log";
            //Create File
            String path = filepath + File.separator + filename;
            File logFile = new File(path);
            //Create dir/file
            try {
                logFile.getParentFile().mkdirs();
                logFile.createNewFile();
            }
            catch (Exception ex){
                ex.printStackTrace();
            }
            return path;
        }
        return null;
    }

}

File: app/src/main/java/net/xcreen/restsms/server/ServerLogging.kt
package net.xcreen.restsms.server

import android.content.Context
import androidx.preference.PreferenceManager
import android.util.Log
import java.io.*
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class ServerLogging(filepath: String, private val context: Context) {
    private val filepath: String = filepath + File.separator + "logs"

    /**
     * Write Log-Entry to File
     * @param type - String Log-Type (error,warning,info)
     * @param message - String Log-Message
     */
    fun log(type: String, message: String) {
        //Stop processing, when logging is disabled
        val sharedPref = PreferenceManager.getDefaultSharedPreferences(context)
        if (sharedPref.getBoolean("disable_logging", false)) {
            return
        }
        //Create and get Log-File
        val path = setupLogging()

        //Create Entry
        val dateFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.getDefault())
        val date = Date()
        val entry = dateFormat.format(date) + " - Type: " + type + " - Message: " + message
        when (type) {
            "error" -> Log.e("Server-Logging", entry)
            "warning" -> Log.w("Server-Logging", entry)
            "debug" -> Log.d("Server-Logging", entry)
            "info" -> Log.i("Server-Logging", entry)
            else -> Log.i("Server-Logging", entry)
        }

        //Write to Log-File
        if (path != null) {
            try {
                FileWriter(path, true).use {
                    fw -> BufferedWriter(fw).use {
                        bw -> PrintWriter(bw).use {
                            out -> out.println(entry)
                        }
                    }
                }
            }
            catch (ex: IOException) {
                ex.printStackTrace()
            }
        }
        else {
            Log.i("Server-Logging", "Failed to write to log!")
        }
    }

    /**
     * Get File for Logging
     * @return String - Log-File-Path
     */
    private fun setupLogging(): String? {
        //Create Filename
        val dateFormat: DateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault())
        val date = Date()
        val filename = dateFormat.format(date) + "-server.log"
        //Create File
        val path = filepath + File.separator + filename
        val logFile = File(path)
        //Create dir/file
        try {
            logFile.parentFile?.mkdirs()
            logFile.createNewFile()
        }
        catch (ex: Exception) {
            ex.printStackTrace()
        }
        return path
    }
}